docs/madr/decisions/006-additional-retry-logic.md

```diff
@@ -0,0 +1,53 @@
+# Support additional Retry options from Envoy
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4494
+
+## Context and Problem Statement
+
+We currently set some defaults for the retryOn field for HTTP in our Retry policy. Some users may need additional flexibility so we should plumb through some extra fields from Envoy.
+
+## Decision Drivers
+
+* User request (additional flexibility)
+
+## Considered Options
+
+1. ProxyTemplate
+2. Adding fields to Retry policy
+
+## Decision Outcome
+
+Chosen option: 2 because we want to provide a lower friction experience to add these fields.
+
+### Policy Changes
+
+The change the policy would add a `retryOn` field to expose additional configuration options beyond our defaults.
+
+```yaml
+spec:
+  conf:
+    http:
+      backOff:
+        baseInterval: 0.025s
+        maxInterval: 0.250s
+      numRetries: 5
+      perTryTimeout: 16s
+      retryOn:
+      - all_5xx
+      - gateway_error
+```
+
+### Potential Implementation Notes
+
+- Due to this field being an enum, we cannot use `5xx` as a field name, so must prefix it (`all_5xx` above) then post 'convert' to the Envoy-native field name.
+- We already use `retriable_status_codes` as a field to specify which codes should be retried, however that is also one of the enum options in `retryOn`. We also need to have a different field name here and post-process back to the Envoy-native field name.
```

Comment:
What about the default value of retryOn:

https://github.com/kumahq/kuma/blob/3fe64a322fa330f28a98cb37d6d56f1713a62e0b/pkg/xds/envoy/listeners/v3/retry_configurer.go#L15-L22

I guess it remains unchanged as long as retryOn is unset and if retryOn is set then it's whatever is in retryOn? 
Please clarify this in there as it's an important compat change

Reply:
Exactly that ^

Reply:
Added section to clarify changes to default behavior.